internal/services/batch/batch_application_data_source.go

```diff
@@ -0,0 +1,86 @@
+package batch
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/batch/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func dataSourceBatchApplication() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceBatchApplicationRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.ApplicationName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"account_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.AccountName,
+			},
+
+			"allow_updates": {
+				Type:     pluginsdk.TypeBool,
+				Computed: true,
+			},
+
+			"default_version": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"display_name": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceBatchApplicationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Batch.ApplicationClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
```

Comment:
Any references to resource group, account name or name can be replaced with the values from the id
```suggestion
	resp, err := client.Get(ctx, id.ResourceGroup id.BatchAccountName, id.Name)
```